app/src/main/java/io/apicurio/registry/rest/v2/SearchResourceImpl.java

```diff
@@ -102,26 +101,42 @@ public ArtifactSearchResults searchArtifacts(String name, Integer offset, Intege
 
         Set<SearchFilter> filters = new HashSet<SearchFilter>();
         if (!StringUtil.isEmpty(name)) {
-            filters.add(new SearchFilter(SearchFilterType.name, name));
+            filters.add(SearchFilter.ofName(name));
         }
         if (!StringUtil.isEmpty(description)) {
-            filters.add(new SearchFilter(SearchFilterType.description, description));
+            filters.add(SearchFilter.ofDescription(description));
         }
         if (!StringUtil.isEmpty(group)) {
-            filters.add(new SearchFilter(SearchFilterType.group, gidOrNull(group)));
+            filters.add(SearchFilter.ofGroup(gidOrNull(group)));
         }
 
         if (labels != null && !labels.isEmpty()) {
-            labels.forEach(label -> filters.add(new SearchFilter(SearchFilterType.labels, label)));
+            labels.forEach(label -> filters.add(SearchFilter.ofLabel(label)));
         }
         if (properties != null && !properties.isEmpty()) {
-            properties.forEach(label -> filters.add(new SearchFilter(SearchFilterType.properties, label)));
+            properties.stream()
+                .map(prop -> {
+                   int delimiterIndex = prop.indexOf(":");
+                   if (delimiterIndex < 0) {
+                       throw new BadRequestException("property search filter wrong formatted, missing ':' delimiter");
+                   }
+                   if (delimiterIndex == 0) {
+                       throw new BadRequestException("property search filter wrong formatted, missing left side of  ':' delimiter");
+                   }
+                   if (delimiterIndex == (prop.length() - 1)) {
+                       throw new BadRequestException("property search filter wrong formatted, missing right side of  ':' delimiter");
+                   }
+                   String propertyKey = prop.substring(0, delimiterIndex);
+                   String propertyValue = prop.substring(delimiterIndex + 1);
+                   return SearchFilter.ofProperty(propertyKey, propertyValue);
+                })
+                .forEach(filters::add);
```

Comment:
I guess here is where we would need to support searching just by the presence of a property.  I will merge this PR and open a new GH issue though.